### PR TITLE
tre-command: init at 0.2.2

### PR DESCRIPTION
**Pull request by dduan**
###### Motivation for this change
tre is a improved version of the command `tree`. It's main additions:

1. colored output
2. ignores paths specified in .gitignore
3. editor alias for each entity listed

Source/Homepage: https://github.com/dduan/tre

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

I'd like to be maintainer of this package for Nix. (I'm the author of the CLI tool).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1563,6 +1563,12 @@
     githubId = 14032;
     name = "Daniel Brockman";
   };
+  dduan = {
```
Comment (mmahut) [inline maintainers/maintainer-list.nix]: Please include this a separate commit.

File: pkgs/tools/system/tre/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+with import <nixpkgs> {};
+
+rustPlatform.buildRustPackage rec {
+  name = "tre-${version}";
```
Comment (mmahut) [inline pkgs/tools/system/tre/default.nix]: Please use ```pname```.

File: pkgs/tools/system/tre/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+with import <nixpkgs> {};
```
Comment (mmahut) [inline pkgs/tools/system/tre/default.nix]: Please don't import all the packages.

File: pkgs/tools/system/tre/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "${version}";
+    sha256 = "1fazw2wn738iknbv54gv7qll7d4q2gy9bq1s3f3cv21cdv6bqral";
+  };
+
+  cargoSha256 = "1lf7g96jh7apw537a0sm3a1ivf36bbxqbzllldjkbrh0x0fb4wpb";
+  verifyCargoDeps = true;
+
+  meta = with stdenv.lib; {
+    description = "Tree command, improved";
+    homepage = https://github.com/dduan/tre;
```
Comment (mmahut) [inline pkgs/tools/system/tre/default.nix]: Please quote the url.

**Comment by dduan** [comment]
@mmahut thanks for your review! I've addressed your feedbacks.

File: pkgs/tools/system/tre/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ rustPlatform, fetchFromGitHub, stdenv }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "tre-${version}";
```
Comment (mmahut) [inline pkgs/tools/system/tre/default.nix]:
```suggestion
  pname = "tre";
```

The name is contrusted of ```${pname}-${version}``` so you should use just the name itself.

File: pkgs/tools/system/tre/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  version = "v0.2.2";
```
Comment (mmahut) [inline pkgs/tools/system/tre/default.nix]:
```suggestion
  version = "0.2.2";
```

Please move the ```v``` in the tag to rev, as noted below.

File: pkgs/tools/system/tre/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "dduan";
+    repo = "tre";
+    rev = "${version}";
```
Comment (mmahut) [inline pkgs/tools/system/tre/default.nix]:
```suggestion
    rev = "v${version}";
```

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -21304,6 +21304,8 @@ in
 
   tdrop = callPackage ../applications/misc/tdrop { };
 
+  tre-command = callPackage ../tools/system/tre {};
```
Comment (mmahut) [inline pkgs/top-level/all-packages.nix]: Any reason for renaming this?

Comment (dduan) [reply pkgs/top-level/all-packages.nix]: yes, there's a existing library named `tre` here already.

Comment (mmahut) [reply pkgs/top-level/all-packages.nix]: Ah, I see, can we rename the directory too in that case (pname can stay as it is)?

Comment (dduan) [reply pkgs/top-level/all-packages.nix]: done

**Comment by mmahut** [comment]
@GrahamcOfBorg build  tre-command

File: pkgs/top-level/all-packages.nix (hunk 8 above)
Comment (mmahut) [reply pkgs/top-level/all-packages.nix]: I think you need to change the path here too.

Comment (dduan) [reply pkgs/top-level/all-packages.nix]: Yup done

**Comment by dduan** [comment]
@mmahut lemme know if there's further improvement to be made in order for this to be merged. Thanks for your help so far!

**Comment by mmahut** [comment]
@GrahamcOfBorg build tre-command

**Comment by mmahut** [comment]
Looks good to me. Thank you Daniel for your contribution and welcome! :)

**Pull request merged**